### PR TITLE
fix: handling functions update flow errors when no functions are present

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of how you validated changes

* tested manually

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts [hunk 1]
```diff
@@ -141,14 +141,22 @@ export async function updateWalkthrough(context, lambdaToUpdate?: string) {
     .map(resource => resource.resourceName);
 
   if (lambdaFuncResourceNames.length === 0) {
-    context.print.error('No Lambda function resource to update. Use "amplify add function" to create a new function.');
-    return;
+    const errMessage = 'No Lambda function resource to update. Use "amplify add function" to create a new function.';
+    const error = new ResourceDoesNotExistError(errMessage);
+    error.name = 'FunctionResourceNotFound';
```
Comment (cjihrig) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts:146]: Setting the error's name here and on line 156 is a little confusing. From what I can tell by scanning the codebase, it looks like we only overwrite the name on builtin `Error` objects.

File: packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts [hunk 2]
```diff
@@ -217,12 +217,27 @@ export async function updateFunctionResource(context, category, service, paramet
...
+      parameters = await serviceConfig.walkthroughs.updateWalkthrough(context, parameters, resourceToUpdate);
+      if (parameters.dependsOn) {
+        context.amplify.updateamplifyMetaAfterResourceUpdate(category, parameters.resourceName, 'dependsOn', parameters.dependsOn);
+      }
+      saveMutableState(parameters);
+      saveCFNParameters(parameters);
+    } catch (e) {
+      if (e.name === 'FunctionResourceNotFound' || e.name === 'SelectedLambdaFunctionNotFound') {
+        // exit here as for these errors function parameters wont be defined
+        context.print.warning(e.message);
```
Comment (cjihrig) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts:230]: No need to address it in this PR, but at some point, we should probably DRY our error handling throughout the codebase a bit 😄 .

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts (hunk 1 above)
Comment (cjihrig) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts:146]: Ping @akshbhu

**Comment by akshbhu** [comment]
Closing this PR

**Pull request closed without merging**